src/plugins/search/plugin.search.js

```diff
@@ -199,11 +199,22 @@ BookReader.prototype.search = function(term = '', overrides = {}) {
  */
 BookReader.prototype.BRSearchCallback = function(results, options) {
   this.searchResults = results;
+  console.log('11options.goToFirstResult:- ', options.goToFirstResult);
 
   this.updateSearchHilites();
   this.removeProgressPopup();
   if (options.goToFirstResult) {
-    this._searchPluginGoToResult(results.matches[0].par[0].page);
+    console.log('results.matches[0].par[0].page', results.matches[0].par[0].page)
+    let index = results.matches[0].par[0].page;
+    this._searchPluginGoToResult((index % 2 === 0) ? index : --index);
```

Comment:
so this one is intriguing, 

apparently `results.matches[0].par[0].page` maps to => `leafNum` not index.

the image1 below has results on left & page data by indices (152, 153) on the right.

a few notes:
- the change above results in the correct index of `152` = leafNum `153` ([search results return](https://ia802806.us.archive.org/fulltext/inside.php?item_id=threehundredyear00cors_0&doc=threehundredyear00cors_0&path=/33/items/threehundredyear00cors_0&q=mary+mathis) )
- the fn that takes a user to the first result is [here](https://github.com/internetarchive/bookreader/blob/master/src/plugins/search/plugin.search.js#L349). => it uses `br._models.book.getPage(pageIndex)` (uses index, not leafNumber)
- **QA Fails** with this search, 1up mode: `https://ia-petabox-neeraj.archive.org/details/threehundredyear00cors_0/mode/1up?q=sommers+carson`

A possible solution is just to reduce the index by 1. 
_IF_ we go with `--page` to get index from leafNum, we [may] need to confirm that leafNum is always +1 from index? 

I'd prefer to use leafNum to turn the page so that we can just rely on the data...but that is more work.

@cdrini, @nsharma123 what do you think - should we lean on reducing results index by 1 OR get page info by leaf num OR or be able to turn page by LeafNum OR something else...?

img 1:
<img width="1071" alt="image" src="https://user-images.githubusercontent.com/7840857/119405591-58734e00-bc96-11eb-84b8-c0da0da95350.png">

Reply:
@nsharma123 from our slack discussion thread with @cdrini, let's update `_searchPluginGoToResult` to use `book.getPage(book.leafNumToIndex(leafNum))` somewhere around line 349